src/sql/src/plan/statement/ddl.rs

```diff
@@ -1720,6 +1722,26 @@ pub fn plan_create_table_from_source(
         source_connection,
     )?;
 
+    let name = scx.allocate_qualified_name(normalize::unresolved_item_name(name.clone())?)?;
+
+    // Allow users to specify a timeline. If they do not, determine a default
+    // timeline for the source.
+    let timeline = match timeline {
+        None => match envelope {
+            SourceEnvelope::CdcV2 => {
+                Timeline::External(scx.catalog.resolve_full_name(&name).to_string())
+            }
+            _ => Timeline::EpochMilliseconds,
+        },
+        // TODO(benesch): if we stabilize this, can we find a better name than
+        // `mz_epoch_ms`? Maybe just `mz_system`?
+        Some(timeline) if timeline == "mz_epoch_ms" => Timeline::EpochMilliseconds,
+        Some(timeline) if timeline.starts_with("mz_") => {
+            return Err(PlanError::UnacceptableTimelineName(timeline));
+        }
+        Some(timeline) => Timeline::User(timeline),
+    };
```

Comment:
this is the same logic as already exists in `plan_create_source`. It will be removed from that function once we drop support for outputting data to the primary source collection